crates/dbsp/src/circuit/dbsp_handle.rs

```diff
@@ -543,6 +554,18 @@ impl DBSPHandle {
         Ok(dir_path)
     }
 
+    pub fn retrieve_profile(&mut self) -> Result<DbspProfile, DBSPError> {
+        let mut profiles = vec![Default::default(); self.status_receivers.len()];
+
+        self.broadcast_command(Command::RetrieveProfile, |worker, resp| {
+            if let Response::Profile(prof) = resp {
```

Comment:
What if not all show up?

Reply:
This will deadlock. The same is true for `step()`.